### DISCUSSION TITLE
Users unable to see servers

### DISCUSSION THREAD

**Discussion by JMG81** [gravitational/teleport#14008] [Q&A] [answered]
Greetings, new to teleport and impressed so far.  Doing a trial run with open source at the moment (Teleport v9.3.7 git:v9.3.7-0-g5dc5dac go1.17.11) to see if it will meet our needs.  Anyway, newly created users in the GUI aren't able to see the server list when logged in.  Under the roles assigned, there is:

    logins:
    - '{{internal.logins}}'

In the docs, it says "If no OS login is specified, it defaults to the same name as the Teleport user".  So I would think the user would be able to attempt a login to servers as their user name.  Docs that I am referring to are here:  https://goteleport.com/docs/setup/admin/users/

I am able to edit the user's .yaml after creation from the GUI and specify the logins there, then users are able to see servers and login as their username.  I can also add a login to the role, such as ubuntu, and servers will be visible but only with the option to login as ubuntu.  Perhaps I am missing something?

**Comment by webvictim** [comment, marked as answer]
This is perhaps confusing, but to be clear - `{{internal.logins}}` actually expands out to any `traits` which are set on the user object itself. So if you wanted to allow each user to use their own individual login (rather than a shared login in a role like `ubuntu`), you can set that in their traits.

`webvictim.yaml`:
```yaml
kind: user
metadata:
  name: webvictim
spec:
  roles:
  - access
  - editor
  traits:
    logins:
    - webvictim
```

Then `tctl create -f webvictim.yaml` to configure the user.

Unfortunately there isn't a way to do this automatically in Teleport OSS. The Enterprise edition provides better ways to set a claim/attribute from SSO as a possible username inside a role rather than needing to do it for individual users.

The text from the docs (`If no OS login is specified, it defaults to the same name as the Teleport user`) means that if you're logged into your local machine as `john` and you run `tsh ssh server` without providing a username, it will actually run `tsh ssh john@server`.

**Comment by JMG81** [comment]
Thank you much for the detailed explanation, it makes perfect sense.
